src/platform/APM32/mk/APM32F4.mk

```diff
@@ -135,59 +135,59 @@ DEVICE_FLAGS    = -DUSE_DAL_DRIVER -DHSE_VALUE=$(HSE_VALUE) -DAPM32
 ifeq ($(TARGET_MCU),APM32F405xx)
 DEVICE_FLAGS    += -DAPM32F405xx
 LD_SCRIPT       = $(LINKER_DIR)/apm32_flash_f405.ld
-STARTUP_SRC     = apm32/startup_apm32f405xx.S
+STARTUP_SRC     = startup/startup_apm32f405xx.S
 MCU_FLASH_SIZE  := 1024
 
 else ifeq ($(TARGET_MCU),APM32F407xx)
 DEVICE_FLAGS    += -DAPM32F407xx
 LD_SCRIPT       = $(LINKER_DIR)/apm32_flash_f407.ld
-STARTUP_SRC     = apm32/startup_apm32f407xx.S
+STARTUP_SRC     = startup/startup_apm32f407xx.S
 MCU_FLASH_SIZE  := 1024
 else
-$(error TARGET_MCU [$(TARGET_MCU] is not supported)
+$(error TARGET_MCU [$(TARGET_MCU)] is not supported)
 endif
 
 MCU_COMMON_SRC = \
-            startup/apm32/system_apm32f4xx.c \
+            startup/system_apm32f4xx.c \
             drivers/inverter.c \
             drivers/dshot_bitbang_decode.c \
             drivers/pwm_output_dshot_shared.c \
-            drivers/mcu/apm32/bus_spi_apm32.c \
-            drivers/mcu/apm32/bus_i2c_apm32.c \
-            drivers/mcu/apm32/bus_i2c_apm32_init.c \
-            drivers/mcu/apm32/camera_control.c \
-            drivers/mcu/apm32/debug.c \
-            drivers/mcu/apm32/dma_reqmap_mcu.c \
-            drivers/mcu/apm32/dshot_bitbang.c \
-            drivers/mcu/apm32/dshot_bitbang_ddl.c \
-            drivers/mcu/apm32/eint_apm32.c \
-            drivers/mcu/apm32/io_apm32.c \
-            drivers/mcu/apm32/light_ws2811strip_apm32.c \
-            drivers/mcu/apm32/persistent_apm32.c \
-            drivers/mcu/apm32/pwm_output_apm32.c \
-            drivers/mcu/apm32/pwm_output_dshot_apm32.c \
-            drivers/mcu/apm32/rcm_apm32.c \
-            drivers/mcu/apm32/serial_uart_apm32.c \
-            drivers/mcu/apm32/timer_apm32.c \
-            drivers/mcu/apm32/transponder_ir_io_apm32.c \
-            drivers/mcu/apm32/timer_apm32f4xx.c \
-            drivers/mcu/apm32/adc_apm32f4xx.c \
-            drivers/mcu/apm32/dma_apm32f4xx.c \
-            drivers/mcu/apm32/serial_uart_apm32f4xx.c \
-            drivers/mcu/apm32/system_apm32f4xx.c
+            bus_spi_apm32.c \
```

Comment:
Could we preserve some relation to drivers ?

Reply:
Not really, as the file is no longer located in "drivers", we could create a directory in the platform location called drivers also, but at this point it would be superfluous. Remember you are in the make file for the platform.